doc/cascadia/Keybindings-Arguments.md

```diff
@@ -0,0 +1,315 @@
+actions are _fundamentally_ the same function. However, we've needed to define 9
+different actions to enable the user to provide different values to the `newTab`
+function.
+
+With this change, we'll be able to remove these _essentially_ duplicated events,
+and allow the user to specify arbitrary arguments to these functions.
+
+## Inspiration
+
+Largely inspired by the keybindings in VsCode and Sublime Text.
+
+## Solution Design
+
+We'll need to introduce args to some actions that we already have defined. These
+are the actions I'm thinking about when writing this spec:
+
+```csharp
+    // These events already exist like this:
+    delegate void NewTabWithProfileEventArgs(Int32 profileIndex);
+    delegate void SwitchToTabEventArgs(Int32 profileIndex);
+    delegate void ResizePaneEventArgs(Direction direction);
+    delegate void MoveFocusEventArgs(Direction direction);
+
+    // These events either exist in another form or don't exist.
+    delegate void CopyTextEventArgs(Boolean copyWhitespace);
+    delegate void ScrollEventArgs(Int32 numLines);
+    delegate void SplitProfileEventArgs(Orientation splitOrientation, Int32 profileIndex);
+```
+
+Ideally, after this change, the bindings for these actions would look something
+like the following:
+
+```js
+{ "keys": ["ctrl+shift+1"], "command": "newTabProfile", "args": { "profileIndex":0 } },
+{ "keys": ["ctrl+shift+2"], "command": "newTabProfile", "args": { "profileIndex":1 } },
+// etc...
+
+{ "keys": ["alt+1"], "command": "switchToTab", "args": { "index":0 } },
+{ "keys": ["alt+2"], "command": "switchToTab", "args": { "index":1 } },
+// etc...
+
+{ "keys": ["alt+shift+down"], "command": "resizePane", "args": { "direction":"down" } },
+{ "keys": ["alt+shift+up"], "command": "resizePane", "args": { "direction":"up" } },
+// etc...
+
+{ "keys": ["alt+down"], "command": "moveFocus", "args": { "direction":"down" } },
+{ "keys": ["alt+up"], "command": "moveFocus", "args": { "direction":"up" } },
+// etc...
+
+{ "keys": ["ctrl+c"], "command": "copy", "args": { "copyWhitespace":true } },
+{ "keys": ["ctrl+shift+c"], "command": "copy", "args": { "copyWhitespace":false } },
+
+{ "keys": ["ctrl+shift+down"], "command": "scroll", "args": { "numLines":1 } },
+{ "keys": ["ctrl+shift+up"], "command": "scroll", "args": { "numLines":-1 } },
+
+{ "keys": ["ctrl+alt+1"], "command": "splitProfile", "args": { "orientation":"vertical", "profileIndex": 0 } },
+{ "keys": ["ctrl+alt+shift+1"], "command": "splitProfile", "args": { "orientation":"horizontal", "profileIndex": 0 } },
+{ "keys": ["ctrl+alt+2"], "command": "splitProfile", "args": { "orientation":"vertical", "profileIndex": 1 } },
+{ "keys": ["ctrl+alt+shift+2"], "command": "splitProfile", "args": { "orientation":"horizontal", "profileIndex": 1 } },
+// etc...
+```
+
+Note that instead of having 9 different `newTabProfile<N>` actions, we have a
+singular `newTabProfile` action, and that action requires a `profileIndex` in

```

Comment:
Huh. It makes sense that it could alternatively have a GUID. But then how would we handle if _both_ existed in the `args`? I guess we can't really have an optional member, now can we?